pkg/utils/kubernetes/crd.go

```diff
@@ -75,3 +82,35 @@ func WaitUntilCRDManifestsDestroyed(ctx context.Context, c client.Client, crdNam
 	}
 	return flow.Parallel(fns...)(ctx)
 }
+
+var (
+	crdScheme  *runtime.Scheme
+	crdCodec   runtime.Codec
+	initOnceFn sync.Once
+)
+
+func initializeCRDSchemeAndCodec() {
+	crdScheme = runtime.NewScheme()
+	utilruntime.Must(apiextensionsv1.AddToScheme(crdScheme))
+	ser := json.NewSerializerWithOptions(json.DefaultMetaFactory, crdScheme, crdScheme, json.SerializerOptions{
+		Yaml:   true,
+		Pretty: false,
+		Strict: false,
+	})
+	versions := schema.GroupVersions([]schema.GroupVersion{apiextensionsv1.SchemeGroupVersion})
+	crdCodec = serializer.NewCodecFactory(crdScheme).CodecForVersions(ser, ser, versions, versions)
+}
+
+// DecodeCRD decodes a CRD from a YAML string.
+func DecodeCRD(crdYAML string) (*apiextensionsv1.CustomResourceDefinition, error) {
```

Comment:
Is it possible to write unit tests for this helper func? When invalid YAML is passed, appropriate error will be returned?

Reply:
Added unit tests for `DecodeCRD` function.